effect/shared/src/main/scala/com/monovore/decline/effect/CommandIOApp.scala

```diff
@@ -1,21 +1,26 @@
 package com.monovore.decline.effect
 
-import cats.effect.{IO, IOApp, ExitCode}
+import cats.effect.implicits._
+import cats.effect.{Effect, ExitCode, IO, IOApp}
 import cats.implicits._
-
 import com.monovore.decline._
 
-abstract class CommandIOApp(
+import scala.language.higherKinds
+
+abstract class CommandIOApp[F[_]: Effect](
     name: String,
     header: String,
     helpFlag: Boolean = true,
     version: String = ""
 ) extends IOApp {
 
-  def main: Opts[IO[ExitCode]]
+  def main: Opts[F[_]]
```

Comment:
this destroys the ability of users to set exitcodes, no? This seems an orthogonal change.

Reply:
Well, somehow I missed the public apply method for ExitCode and thought it was only possible to have ExitCode.Success and ExitCode.Error, which turns out to be wrong after a double check.
The original idea was not to force ZIO and Monix users to provide a cats-effect result (ExitCode), but the only other way I see to preserve the ability to explicitly set the exit code is via an Int. 
Using ExitCode is not that dramatic though, I think, so I am fine either rolling back to using it in the return type or using a plain Int, just let me know what you think about and I will push an update.

Reply:
if you are using this and ZIO, you are already depending on cats-effect (since this code does). I think using a type (ExitCode) is better than using Int. Not to mention, using `Int` would be a binary compatible (since this is erased) but could be a runtime error if users link against the wrong version.

I think using ExitCode here (not making a change) is far preferable.

Reply:
Very good points! I pushed a fix